src/interpreter/executors/main.rs

```diff
@@ -226,6 +113,153 @@ where
             }
         }
     }
+}
+
+impl<S, Tx> Interpreter<S, Tx>
+where
+    S: InterpreterStorage,
+{
+    fn _deploy(&mut self, mut create: Create, initial_balances: InitialBalances) -> Result<Create, InterpreterError> {
+        let salt = create.salt();
+        let storage_slots = create.storage_slots();
+        let contract = Contract::try_from(&create)?;
+        let root = contract.root();
+        let storage_root = Contract::initial_state_root(storage_slots.iter());
+        let id = contract.id(salt, &root, &storage_root);
+
+        if !create
+            .outputs()
+            .iter()
+            .any(|output| matches!(output, Output::ContractCreated { contract_id, state_root } if contract_id == &id && state_root == &storage_root))
+        {
+            return Err(InterpreterError::Panic(PanicReason::ContractNotInInputs));
+        }
+
+        let storage = &mut self.storage;
+        storage
+            .deploy_contract_with_id(salt, storage_slots, &contract, &root, &id)
+            .map_err(InterpreterError::from_io)?;
+
+        // TODO: Maybe the code below is useless because we can't spend any gas and can't execute
```

Comment:
I'm not sure do we really need this for the `Create` transaction or not. I tried to have the same code that was before the refactoring, but if you agree that we don't need it, I will remove it in this PR=)

Reply:
Probably a good call to just maintain the existing behavior, but in theory the gas limit should only apply to scripts.

Reply:
I prefer to remove it now if we agree that it is only applicable for `Script`, but I also can create a follow-up PR for it=)

Reply:
Well in the case a user sets a high gas limit, we should still refund it if no gas is consumed.

Reply:
Yeah, but for the `Create` transaction, the `Output` by default can be initially fully refunded because we always spent zero gas=)

Reply:
yeah makes sense, as long as the gas / refund behavior is the same we can refactor this code

Reply:
Removed in this pr then: https://github.com/FuelLabs/fuel-vm/pull/235/commits/b94c69b8519977f879b8bddac909a88221a8c179